src/ui/reducers/app.ts

```diff
@@ -76,11 +76,11 @@ export default function update(
     }
 
     case "set_expected_error": {
-      return { ...state, expectedError: action.error };
+      return { ...state, expectedError: action.error, modal: null };
```

Comment:
There's also `modalOptions` that we probably should clear out just to keep everything tidy.